packages/core/src/external/surescripts/client.ts

```diff
@@ -0,0 +1,365 @@
+    }
+
+    const content = await this.s3.downloadFile({
+      bucket: this.bucket,
+      key: s3Key,
+    });
+
+    if (!dryRun) {
+      const sftpFileName = getSftpFileName(OUTGOING_NAME, fileName);
+      await this.write(sftpFileName, content);
+    }
+    return content;
+  }
+
+  async checkFileStatusWithSurescripts({
+    cxId,
+    fileName,
+    timestamp,
+  }: {
+    cxId: string;
+    fileName: string;
+    timestamp: number;
+  }): Promise<{ didSend: boolean; didVerify: boolean; didReceive: boolean }> {
+    const expectedFileNameInHistory = getSftpFileName(HISTORY_NAME, `${fileName}.${this.senderId}`);
+    const didSend = await this.exists(expectedFileNameInHistory);
+    const didVerify = didSend ? (await this.findVerificationFileName(fileName)) != null : false;
+    const didReceive = didVerify
+      ? (await this.findFlatFileResponseName(cxId, timestamp)) != null
+      : false;
+
+    return {
+      didSend,
+      didVerify,
+      didReceive,
+    };
+  }
+
+  async didCopyPatientLoadFileToSurescripts(transmission: Transmission<TransmissionType>) {
+    const fileName = this.getExpectedPatientLoadFileNameInAuditLogs(
+      transmission.id,
+      transmission.timestamp,
+      transmission.compression
+    );
+    const sftpFileName = getSftpFileName(HISTORY_NAME, fileName);
+    const exists = await this.exists(sftpFileName);
+    return exists;
+  }
+
+  async didReceiveVerificationResponseFromSurescripts(
+    transmission: Transmission<TransmissionType>
+  ) {
+    const fileName = await this.findVerificationFileName(transmission.requestFileName);
+    return fileName != null;
+  }
+
+  async didReceiveFlatFileResponseFromSurescripts(transmission: Transmission) {
+    const fileName = await this.findFlatFileResponseName(transmission.cxId, transmission.timestamp);
+    return fileName != null;
+  }
+
+  private getPatientLoadFileName(
+    id: string, // the UUID of the transmission
+    timestamp: number,
+    compression = true
+  ): string {
+    return [
+      "Metriport_PMA_",
+      dayjs(timestamp).format("YYYYMMDD"),
+      "-",
+      id,
+      compression ? ".gz" : "",
+    ].join("");
+  }
+
+  private getExpectedPatientLoadFileNameInAuditLogs(
+    id: string,
+    timestamp: number,
+    compression = true
+  ) {
+    const fileName = this.getPatientLoadFileName(id, timestamp, compression);
+    return `${fileName}.${this.senderId}`;
+  }
+
+  private async findVerificationFileName(requestFileName: string): Promise<string | undefined> {
+    const requestFileNameWithoutExtension = requestFileName.replace(/\.gz$/, "");
+
+    const results = await this.list(getSftpDirectory(INCOMING_NAME), info => {
+      const parsedFileName = this.parseVerificationFileName(info.name);
+      return (
+        parsedFileName != null && parsedFileName.requestFileName === requestFileNameWithoutExtension
+      );
+    });
+    return results[0];
+  }
+
+  private parseVerificationFileName(remoteFileName: string): {
```

Comment:
Here is the Surescripts specification of the verification file name, for context.

<img width="821" alt="verification-file-naming" src="https://github.com/user-attachments/assets/b60dd382-2b6b-4fae-9ded-85c12bcd82b5" />